pkg/storage/store_pool.go

```diff
@@ -327,6 +327,20 @@ func (sp *StorePool) deadReplicasGossipUpdate(_ string, content roachpb.Value) {
 	detail.deadReplicas = deadReplicas
 }
 
+// updateLocalStoreAfterRebalance is used to update local copy of target store
+// after we make an rebalance immediately.
+func (sp *StorePool) updateLocalStoreAfterRebalance(storeID roachpb.StoreID, repl *Replica) {
+	sp.detailsMu.Lock()
+	defer sp.detailsMu.Unlock()
+	detail := sp.getStoreDetailLocked(storeID)
+	// update LogicalBytes.
+	detail.desc.Capacity.LogicalBytes += repl.GetMVCCStats().Total()
+	// update WritesPerSecond
+	qps, _ := repl.writeStats.avgQPS()
```

Comment:
Only use the qps returned if the duration returned is greater than `MinStatsDuration`

Reply:
Why should we guarantee this?Our intent is to immediately update the write stats, right?

Reply:
The reason for the check is that taking an average over a very small duration makes large errors pretty likely. I saw issues in my testing that were caused by using the stats returned shortly after a replica was initialized, because the stats returned were often significantly higher than their true steady state value.

There may be other ways to get around this, so suggestions are appreciated. Otherwise, though, I think it's a reasonable practice to wait at least a couple seconds before using these stats.

In almost all cases the duration will have been long enough here, since this code is only executing on the leaseholder.